src/service/src/transport.rs

```diff
@@ -49,14 +49,31 @@ pub struct Client<Out, In> {
 impl<Out: Message, In: Message> Client<Out, In> {
     /// Connect to the server at the given address.
     pub async fn connect(address: &str, version: Version) -> anyhow::Result<Self> {
+        let dest_host = host_from_address(address);
         let stream = Stream::connect(address).await?;
         info!(%address, "ctp: connected to server");
 
-        let conn = Connection::start(stream, version).await?;
+        let conn = Connection::start(stream, version, dest_host).await?;
         Ok(Self { conn })
     }
 }
 
+/// Helper function to extract the host part from an address string.
+///
+/// This function assumes addresses to be of the form `<host>:<port>` or `<protocol>:<host>:<port>`
+/// and yields `None` otherwise.
+fn host_from_address(address: &str) -> Option<String> {
```

Comment:
Would it make sense to have some form of URL abstraction so this logic wouldn't need to live here? Not urgent, but maybe worth recording.

Reply:
Yeah, I also don't love this part. We have a similar thing in `create_sockets`. The issue is that our address format here isn't a URL format, and so parsing it with a crate like `url` doesn't work. The best I can think of is putting some parsing function into `mz_ore::netio`.